trieste/acquisition/optimizer.py

```diff
@@ -197,68 +199,172 @@ def optimize_continuous(
+
+    num_optimization_runs = tf.shape(starting_points)[0].numpy()
+
+    def _objective_value(x: TensorType) -> TensorType:
+        return -target_func(tf.expand_dims(x, 1))  # [len(x),1]
+
+    def _objective_value_and_gradient(x: TensorType) -> Tuple[TensorType, TensorType]:
+        return tfp.math.value_and_gradient(_objective_value, x)  # [len(x), 1], [len(x), D]
+
+    if isinstance(
+        space, TaggedProductSearchSpace
+    ):  # build continuous relaxation of discrete subspaces
+        bounds = [
+            get_bounds_of_box_relaxation_around_point(space, starting_points[i : i + 1])
+            for i in tf.range(num_optimization_runs)
+        ]
+    else:
+        bounds = [spo.Bounds(space.lower, space.upper)] * num_optimization_runs
+
+    # Initialize the numpy arrays to be passed to the greenlets
+    np_batch_x = np.zeros((num_optimization_runs, tf.shape(starting_points)[1]), dtype=np.float64)
+    np_batch_y = np.zeros((num_optimization_runs,), dtype=np.float64)
+    np_batch_dy_dx = np.zeros(
+        (num_optimization_runs, tf.shape(starting_points)[1]), dtype=np.float64
+    )
+
+    # Set up child greenlets
+    child_greenlets = [ScipyLbfgsBGreenlet() for _ in range(num_optimization_runs)]
+    child_results: List[Union[spo.OptimizeResult, np.ndarray]] = [
+        gr.switch(starting_points[i].numpy(), bounds[i], optimizer_args)
+        for i, gr in enumerate(child_greenlets)
+    ]
+
+    while True:
+        all_done = True
+        for i, result in enumerate(child_results):  # Process results from children.
+            if isinstance(result, spo.OptimizeResult):
+                continue  # children return a `spo.OptimizeResult` if they are finished
+            all_done = False
+            assert isinstance(result, np.ndarray)  # or an `np.ndarray` with the query `x` otherwise
+            np_batch_x[i, :] = result
+
+        if all_done:
+            break
+
+        # Batch evaluate query `x`s from all children.
+        batch_x = tf.constant(np_batch_x, dtype=tf_dtype)  # [num_optimization_runs, d]
+        batch_y, batch_dy_dx = _objective_value_and_gradient(batch_x)
+        np_batch_y = batch_y.numpy().astype("float64")
+        np_batch_dy_dx = batch_dy_dx.numpy().astype("float64")
+
+        for i, greenlet in enumerate(child_greenlets):  # Feed `y` and `dy_dx` back to children.
+            if greenlet.dead:  # Allow for crashed greenlets
+                continue
+            child_results[i] = greenlet.switch(np_batch_y[i], np_batch_dy_dx[i, :])
+
+    best_run_id = np.argmax([-result.fun for result in child_results])  # identify best solution
+    np_chosen_point = child_results[best_run_id].x.reshape(1, -1)  # [1, D]
+    success = np.any(
+        [result.success for result in child_results]
+    )  # Check that at least one optimization was successful
+
+    return tf.constant(np_chosen_point, dtype=tf_dtype), success  # convert back to TF
```

Comment:
I'm thinking that some other metadata might be useful as well, e.g. which k was it that yielded the best solution, what were the values and points in all k runs, did we enter the recovery runs etc - perhaps logging this to the tensorboard? 

I think it would be useful for monitoring, e.g. to spot whether you might need to decrease/increase the number of runs, or if you are consistently entering recovery runs that something might be off

Reply:
Yep I agree. Will save for another PR! https://github.com/secondmind-labs/trieste/issues/440